test/connectors/util/CustomHttpReaderSpec.scala

```diff
@@ -30,18 +30,22 @@ class CustomHttpReaderSpec extends AnyFreeSpec with Matchers with OptionValues w
 
   val _2xxGenerator: Gen[Int] = Gen.oneOf(Seq(Status.OK, Status.CREATED, Status.ACCEPTED, Status.NO_CONTENT))
   val clientErrorGenerator: Gen[Int] = Gen.oneOf(Seq(Status.LOCKED, Status.FORBIDDEN, Status.NOT_FOUND, Status.BAD_REQUEST))
-  val serverErrorGenerator: Gen[Int] = Gen.oneOf(Seq(Status.INTERNAL_SERVER_ERROR, Status.NOT_IMPLEMENTED, Status.SERVICE_UNAVAILABLE, Status.GATEWAY_TIMEOUT))
+  val serverErrorGenerator: Gen[Int] = Gen.oneOf(Seq(Status.NOT_IMPLEMENTED, Status.SERVICE_UNAVAILABLE, Status.BAD_GATEWAY))
```

Comment:
It might be best to update the generators and use Gen.between for generating all 2xx, 4xx and 5xx status codes and then filtering out the codes we convert or handle.

Reply:
I can appreciate why you'd suggest using between, but these are meant to explicitly exclude some values - the ones we map. So unless Between can also include exception values, or we filter those out, it'll just cause the tests to fail.